en/02_Developer_Guides/08_Performance/06_ORM.md

```diff
@@ -72,3 +75,250 @@ SilverStripe\ORM\FieldType\DBPolymorphicForeignKey:
+
+The main limitation is that filtering or sorting an `EagerLoadedList` will be done in PHP rather than as part of the database query, since we have already loaded all its relevant data into memory.
+
+> [!WARNING]
+> `EagerLoadedList` can't filter or sort by fields on relations using dot notation (e.g. `sort('MySubRelation.Title')` won't work).
+>
+> Manipulating the eager loading query is significantly (up to ~600%) faster than Filtering an `EagerLoadedlist` after the query has been made.
+
+You can avoid those problems by applying manipulations such as filtering and sorting to the eager loading query as part of your call to `eagerLoad()`.
+
+You can pass an associative array into the `eagerLoad()` method, with relation chains as the keys and callbacks as the values. The callback accepts a `DataList` argument, and must return a `DataList`.
+
+> [!CAUTION]
+> You can't manipulate the lists of `has_one` or `belongs_to` relations. This functionality is intended primarily as a way to pre-filter or pre-sort `has_many` and `many_many` relations.
+
+```php
+use SilverStripe\ORM\DataList;
+
+$teams = Team::get()->eagerLoad([
+    'Players' => fn (DataList $list) => $list->filter(['Age:GreaterThan' => 18]),
+]);
+```
+
+The list passed into your callback represents the query for that relation on *all* of the records you're fetching. For example, the `$list` variable above is a `DataList` that will fetch all `Player` records in the `Players` relation for all `Team` records (so long as they match the filter applied in the callback).
+
+Note that each relation in the relation chain (e.g. `Players`, `Players.Fans`, `Players.Fans.Events`) can have their own callback:
+
+```php
+use SilverStripe\ORM\DataList;
+
+$teams = Team::get()->eagerLoad([
+    'Players' => fn (DataList $list) => $list->filter(['Age:GreaterThan' => 18]),
+    'Players.Fans' => fn (DataList $list) => $list->filter(['Name:PartialMatch:nocase' => 'Sam']),
+    'Players.Fans.Events' => fn (DataList $list) => $list->filter(['DollarCost:LessThan' => 200]),
+]);
+```
+
+If you have complex branching logic, and in some branches you want to avoid pre-filtering or perform different filtering, you can declare a different callback (or `null`) for that relation chain by calling `eagerLoad()` again. Note that subsequent callbacks *replace* previous callbacks.
+
+```php
+use SilverStripe\ORM\DataList;
+
+$teams = Team::get()->eagerLoad([
+    // Remove any callback that was previously defined for this relation chain
+    'Players' => fn (DataList $list) => null,
+    // Replace any callback that was previously defined for this relation chain.
+    // If you want to apply *additional* filters rather than replacing existing ones, you will
+    // need to declare the original filter again here.
+    'Players.Fans.Events' => fn (DataList $list) => $list->filter(['DollarCost:GreaterThan' => 100]),
+]);
+```
+
+## Making raw SQL queries {#raw-sql}
```

Comment:
Note that this section is new i.e. it's not part of the eager loading section that was moved